syncmaster/backend/providers/auth/dummy_provider.py

```diff
@@ -0,0 +1,118 @@
+from syncmaster.exceptions.auth import AuthorizationError
+
+log = logging.getLogger(__name__)
+
+
+class DummyAuthProvider(AuthProvider):
+    def __init__(
+        self,
+        settings: Annotated[DummyAuthProviderSettings, Depends(Stub(DummyAuthProviderSettings))],
+        unit_of_work: Annotated[UnitOfWork, Depends()],
+    ) -> None:
+        self._settings = settings
+        self._uow = unit_of_work
+
+    @classmethod
+    def setup(cls, app: FastAPI) -> FastAPI:
+        settings = DummyAuthProviderSettings.parse_obj(app.state.settings.auth.dict(exclude={"provider"}))
+        log.info("Using %s provider with settings:\n%s", cls.__name__, pformat(settings))
+        app.dependency_overrides[AuthProvider] = cls
+        app.dependency_overrides[DummyAuthProviderSettings] = lambda: settings
+        return app
+
+    async def get_current_user(self, access_token: str, *args, **kwargs) -> User:
+        if not access_token:
+            raise AuthorizationError("Missing auth credentials")
+
+        user_id = self._get_user_id_from_token(access_token)
+        user = await self._uow.user.read_by_id(user_id)
+        return user
+
+    async def get_token_password_grant(
+        self,
+        grant_type: str | None = None,
+        login: str | None = None,
+        password: str | None = None,
+        scopes: list[str] | None = None,
+        client_id: str | None = None,
+        client_secret: str | None = None,
+    ) -> dict[str, Any]:
+        if not login or not password:
+            raise AuthorizationError("Missing auth credentials")
+
+        log.info("Get/create user %r in database", login)
+        async with self._uow:
+            try:
+                user = await self._uow.user.read_by_username(login)
+            except EntityNotFoundError:
+                user = await self._uow.user.create(
+                    username=login,
+                    email=f"{login}@example.com",
+                    first_name=f"{login}_first",
+                    middle_name=f"{login}_middle",
+                    last_name=f"{login}_last",
+                    is_active=True,
+                )
+
+        log.info("User with id %r found", user.id)
+        if not user.is_active:
+            raise AuthorizationError(f"User {user.username!r} is disabled")
+
+        log.info("Generate access token for user id %r", user.id)
+        access_token, expires_at = self._generate_access_token(user_id=user.id)
+        return {
+            "access_token": access_token,
+            "token_type": "bearer",
+            "expires_at": expires_at,
+        }
+
+    async def get_token_authorization_code_grant(
+        self,
+        code: str,
+        redirect_uri: str,
+        scopes: list[str] | None = None,
+        client_id: str | None = None,
+        client_secret: str | None = None,
+    ) -> dict[str, Any]:
+        raise NotImplementedError("Authorization code grant is not supported by DummyAuthProvider.")
```

Comment:
If DummyAuthProvider does not implement `get_token_authorization_code_grant`, and KeycloakAuthProvider does not implement `get_token_password_grant`, what's the point of hading basic AuthProvider class?

My proposal is to move auth endpoints registration to providers themselves, so each implementation have its own routes and methods. The only common method they have is `get_current_user`

Reply:
Imho it looks strange to move api auth routes to auth providers because it makes code harder to navigate

Reply:
I don't have strong opinion about that, so we could use the current solution